### PR TITLE
Add support for Sway-native types wrapped in custom types

### PR DESCRIPTION
**Pull request by digorithm**
Closes https://github.com/FuelLabs/fuels-rs/issues/326.

This PR adds support for Sway-native types (`ContractId`, `AssetId`, `Address`) inside custom types being passed as input/output in contract calls. Example:

```Rust
let instance = MyContract::new(id.to_string(), wallet.clone());

let user = User {
    weight: 10,
    address: Address::zeroed(),
};
let result = instance.wrapped_address(user).call().await.unwrap();

assert_eq!(result.value.address, Address::zeroed());
```

### CODE REVIEW

File: packages/fuels-core/src/code_gen/custom_types_gen.rs [hunk 1]
```diff
@@ -273,14 +272,14 @@ pub fn expand_custom_enum(name: &str, prop: &Property) -> Result<TokenStream, Er
             #( #enum_variants ),*
         }
 
-        impl #enum_ident {
-            pub fn param_types() -> Vec<ParamType> {
+        impl Parameterize for #enum_ident {
```
Comment (vnepveu) [inline packages/fuels-core/src/code_gen/custom_types_gen.rs:275]: Why is the `Parameterize` trait needed for general enums and structs?

**Review by vnepveu — CHANGES_REQUESTED** [review]
Some nits and questions

File: packages/fuels-abigen-macro/tests/test_projects/sway_native_types/Forc.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[project]
+name = "sway_native_types"
+authors = ["Fuel Labs"]
```
Comment (MujkicA) [inline packages/fuels-abigen-macro/tests/test_projects/sway_native_types/Forc.toml:3]: This seems to be the standard in all other tomls
```suggestion
authors = ["Fuel Labs <contact@fuel.sh>"]
```

**Review by MujkicA — APPROVED** [review]

File: packages/fuels-core/src/code_gen/custom_types_gen.rs (hunk 1 above)
Comment (vnepveu) [reply packages/fuels-core/src/code_gen/custom_types_gen.rs:275]: Ah right, I had forgotten that we were implementing them outside of traits!

**Review by vnepveu — APPROVED** [review]
:shipit:

**Pull request merged**